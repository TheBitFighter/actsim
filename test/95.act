deftype bar(int<1> a){}

deftype foo(bar a[2]){}

function baz(foo f) : int
{
  chp {
    [ f.a[0].a = 0 -> self := 0
    []else -> self := 1
    ]
  }
}

defproc test ()
{

  int x;
  foo f;

  chp {
    x := baz(f);
    log ("got ", x)
  }
}
test t; 
