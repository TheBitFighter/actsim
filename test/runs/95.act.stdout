[                  10] <>  got 0
